tutorial/tests/testsuite.py

```diff
@@ -1,14 +1,43 @@
 import io
+import pathlib
 from contextlib import redirect_stdout
 from typing import Callable
 
 import pytest
-import ipynbname
 from IPython.core.interactiveshell import InteractiveShell
 from IPython.core.magic import Magics, magics_class, cell_magic
 from IPython.display import HTML, display
 
 
+def _name_from_line(line: str = None):
+    return line.strip().removesuffix(".py") if line else None
+
+
+def _name_from_ipynbname():
+    try:
+        import ipynbname
+        return ipynbname.name()
+    except FileNotFoundError:
+        return None
+
+
+def _name_from_globals():
+    module_path = globals().get('__vsc_ipynb_file__')
+    if module_path:
+        return pathlib.Path(module_path).stem
+    return None
+
+
+def get_module_name(line: str):
```

Comment:
This is just pedantry, but  if you want to use annotations, add the annotation on the return type